From dde02f3383ef2657f9ea45418c31de20c4329756 Mon Sep 17 00:00:00 2001
From: Josef Schlehofer <pepe.schlehofer@gmail.com>
Date: Mon, 19 Apr 2021 09:57:02 +0200
Subject: [PATCH] treewide: reforis-plugins: move reforis-distutils to
 HOST_PYTHON3_PACKAGE_BUILD_DEPENDS

In upcoming OpenWrt 21.02 and OpenWrt master, there was changed a
behavior, I should rather say there was introduced a new feature.
Host packages are defined in packages feed (they don't need to be there
as it supports also relative paths, but leave it for later).
By using defaults, it is in packages feed and in folder
lang/python/host-pip-requirements/foo.txt, where is specified package
from PyPI, its version and most importantly there is a hash of that
package.

So, for host-packages you don't need to create OpenWrt Makefile and have
there only text file and one variable
HOST_PYTHON3_PACKAGE_BUILD_DEPENDS where you define the name (foo) as
defined above.

foo = the package name

Because of that, I removed reforis-distutils package as we don't need to
have separate package as there are no dependencies on the router itself,
but it was just a host package. Since I did it, build system complains
about this harmless warning, however it works and its true:

WARNING: Makefile 'package/feeds/turrispackages/reforis-data-collection-plugin/Makefile' has a build dependency on 'reforis-distutils/host', which does not exist
---
 web/reforis/reforis-data-collection-plugin/Makefile      | 5 ++---
 web/reforis/reforis-diagnostics-plugin/Makefile          | 5 ++---
 web/reforis/reforis-haas-plugin/Makefile                 | 5 ++---
 web/reforis/reforis-netboot-plugin/Makefile              | 5 ++---
 web/reforis/reforis-netmetr-plugin/Makefile              | 5 ++---
 web/reforis/reforis-openvpn-plugin/Makefile              | 5 ++---
 web/reforis/reforis-remote-access-plugin/Makefile        | 5 ++---
 web/reforis/reforis-remote-devices-plugin/Makefile       | 5 ++---
 web/reforis/reforis-remote-wifi-settings-plugin/Makefile | 5 ++---
 web/reforis/reforis-snapshots-plugin/Makefile            | 5 ++---
 web/reforis/reforis-storage-plugin/Makefile              | 5 ++---
 11 files changed, 22 insertions(+), 33 deletions(-)

diff --git a/web/reforis/reforis-data-collection-plugin/Makefile b/web/reforis/reforis-data-collection-plugin/Makefile
index e970438f6..9f167f469 100644
--- a/web/reforis/reforis-data-collection-plugin/Makefile
+++ b/web/reforis/reforis-data-collection-plugin/Makefile
@@ -20,9 +20,8 @@ PKG_MAINTAINER:=CZ.NIC <packaging@turris.cz>
 PKG_LICENSE:=GPL-3.0-or-later
 PKG_LICENSE_FILES:=LICENSE
 
-PKG_BUILD_DEPENDS:=\
-	node/host \
-	reforis-distutils/host \
+HOST_PYTHON3_PACKAGE_BUILD_DEPENDS:=reforis-distutils
+PKG_BUILD_DEPENDS:=node/host
 
 include $(INCLUDE_DIR)/package.mk
 include $(TOPDIR)/feeds/packages/lang/python/python3-package.mk
diff --git a/web/reforis/reforis-diagnostics-plugin/Makefile b/web/reforis/reforis-diagnostics-plugin/Makefile
index a3a833cbf..289f7b696 100644
--- a/web/reforis/reforis-diagnostics-plugin/Makefile
+++ b/web/reforis/reforis-diagnostics-plugin/Makefile
@@ -20,9 +20,8 @@ PKG_MAINTAINER:=CZ.NIC <packaging@turris.cz>
 PKG_LICENSE:=GPL-3.0-or-later
 PKG_LICENSE_FILES:=LICENSE
 
-PKG_BUILD_DEPENDS:=\
-	node/host \
-	reforis-distutils/host \
+HOST_PYTHON3_PACKAGE_BUILD_DEPENDS:=reforis-distutils
+PKG_BUILD_DEPENDS:=node/host
 
 include $(INCLUDE_DIR)/package.mk
 include $(TOPDIR)/feeds/packages/lang/python/python3-package.mk
diff --git a/web/reforis/reforis-haas-plugin/Makefile b/web/reforis/reforis-haas-plugin/Makefile
index c966e706e..e7130e893 100644
--- a/web/reforis/reforis-haas-plugin/Makefile
+++ b/web/reforis/reforis-haas-plugin/Makefile
@@ -20,9 +20,8 @@ PKG_MAINTAINER:=CZ.NIC <packaging@turris.cz>
 PKG_LICENSE:=GPL-3.0-or-later
 PKG_LICENSE_FILES:=LICENSE
 
-PKG_BUILD_DEPENDS:=\
-	node/host \
-	reforis-distutils/host \
+HOST_PYTHON3_PACKAGE_BUILD_DEPENDS:=reforis-distutils
+PKG_BUILD_DEPENDS:=node/host
 
 include $(INCLUDE_DIR)/package.mk
 include $(TOPDIR)/feeds/packages/lang/python/python3-package.mk
diff --git a/web/reforis/reforis-netboot-plugin/Makefile b/web/reforis/reforis-netboot-plugin/Makefile
index e58e0cc1c..1e56adea5 100644
--- a/web/reforis/reforis-netboot-plugin/Makefile
+++ b/web/reforis/reforis-netboot-plugin/Makefile
@@ -20,9 +20,8 @@ PKG_MAINTAINER:=CZ.NIC <packaging@turris.cz>
 PKG_LICENSE:=GPL-3.0-or-later
 PKG_LICENSE_FILES:=LICENSE
 
-PKG_BUILD_DEPENDS:=\
-	node/host \
-	reforis-distutils/host \
+HOST_PYTHON3_PACKAGE_BUILD_DEPENDS:=reforis-distutils
+PKG_BUILD_DEPENDS:=node/host
 
 include $(INCLUDE_DIR)/package.mk
 include $(TOPDIR)/feeds/packages/lang/python/python3-package.mk
diff --git a/web/reforis/reforis-netmetr-plugin/Makefile b/web/reforis/reforis-netmetr-plugin/Makefile
index d4207718f..05857fca9 100644
--- a/web/reforis/reforis-netmetr-plugin/Makefile
+++ b/web/reforis/reforis-netmetr-plugin/Makefile
@@ -20,9 +20,8 @@ PKG_MAINTAINER:=CZ.NIC <packaging@turris.cz>
 PKG_LICENSE:=GPL-3.0-or-later
 PKG_LICENSE_FILES:=LICENSE
 
-PKG_BUILD_DEPENDS:=\
-	node/host \
-	reforis-distutils/host \
+HOST_PYTHON3_PACKAGE_BUILD_DEPENDS:=reforis-distutils
+PKG_BUILD_DEPENDS:=node/host
 
 include $(INCLUDE_DIR)/package.mk
 include $(TOPDIR)/feeds/packages/lang/python/python3-package.mk
diff --git a/web/reforis/reforis-openvpn-plugin/Makefile b/web/reforis/reforis-openvpn-plugin/Makefile
index c623c164c..382594ca0 100644
--- a/web/reforis/reforis-openvpn-plugin/Makefile
+++ b/web/reforis/reforis-openvpn-plugin/Makefile
@@ -20,9 +20,8 @@ PKG_MAINTAINER:=CZ.NIC <packaging@turris.cz>
 PKG_LICENSE:=GPL-3.0-or-later
 PKG_LICENSE_FILES:=LICENSE
 
-PKG_BUILD_DEPENDS:=\
-	node/host \
-	reforis-distutils/host \
+HOST_PYTHON3_PACKAGE_BUILD_DEPENDS:=reforis-distutils
+PKG_BUILD_DEPENDS:=node/host
 
 include $(INCLUDE_DIR)/package.mk
 include $(TOPDIR)/feeds/packages/lang/python/python3-package.mk
diff --git a/web/reforis/reforis-remote-access-plugin/Makefile b/web/reforis/reforis-remote-access-plugin/Makefile
index 06d8433ef..61403f5de 100644
--- a/web/reforis/reforis-remote-access-plugin/Makefile
+++ b/web/reforis/reforis-remote-access-plugin/Makefile
@@ -20,9 +20,8 @@ PKG_MAINTAINER:=CZ.NIC <packaging@turris.cz>
 PKG_LICENSE:=GPL-3.0-or-later
 PKG_LICENSE_FILES:=LICENSE
 
-PKG_BUILD_DEPENDS:=\
-	node/host \
-	reforis-distutils/host \
+HOST_PYTHON3_PACKAGE_BUILD_DEPENDS:=reforis-distutils
+PKG_BUILD_DEPENDS:=node/host
 
 include $(INCLUDE_DIR)/package.mk
 include $(TOPDIR)/feeds/packages/lang/python/python3-package.mk
diff --git a/web/reforis/reforis-remote-devices-plugin/Makefile b/web/reforis/reforis-remote-devices-plugin/Makefile
index c91422a14..e463f4c56 100644
--- a/web/reforis/reforis-remote-devices-plugin/Makefile
+++ b/web/reforis/reforis-remote-devices-plugin/Makefile
@@ -20,9 +20,8 @@ PKG_MAINTAINER:=CZ.NIC <packaging@turris.cz>
 PKG_LICENSE:=GPL-3.0-or-later
 PKG_LICENSE_FILES:=LICENSE
 
-PKG_BUILD_DEPENDS:=\
-	node/host \
-	reforis-distutils/host \
+HOST_PYTHON3_PACKAGE_BUILD_DEPENDS:=reforis-distutils
+PKG_BUILD_DEPENDS:=node/host
 
 include $(INCLUDE_DIR)/package.mk
 include $(TOPDIR)/feeds/packages/lang/python/python3-package.mk
diff --git a/web/reforis/reforis-remote-wifi-settings-plugin/Makefile b/web/reforis/reforis-remote-wifi-settings-plugin/Makefile
index 99c6ff2ed..5303f8462 100644
--- a/web/reforis/reforis-remote-wifi-settings-plugin/Makefile
+++ b/web/reforis/reforis-remote-wifi-settings-plugin/Makefile
@@ -20,9 +20,8 @@ PKG_MAINTAINER:=CZ.NIC <packaging@turris.cz>
 PKG_LICENSE:=GPL-3.0-or-later
 PKG_LICENSE_FILES:=LICENSE
 
-PKG_BUILD_DEPENDS:=\
-	node/host \
-	reforis-distutils/host \
+HOST_PYTHON3_PACKAGE_BUILD_DEPENDS:=reforis-distutils
+PKG_BUILD_DEPENDS:=node/host
 
 include $(INCLUDE_DIR)/package.mk
 include $(TOPDIR)/feeds/packages/lang/python/python3-package.mk
diff --git a/web/reforis/reforis-snapshots-plugin/Makefile b/web/reforis/reforis-snapshots-plugin/Makefile
index 565001f97..8a9c9df2e 100644
--- a/web/reforis/reforis-snapshots-plugin/Makefile
+++ b/web/reforis/reforis-snapshots-plugin/Makefile
@@ -20,9 +20,8 @@ PKG_MAINTAINER:=CZ.NIC <packaging@turris.cz>
 PKG_LICENSE:=GPL-3.0-or-later
 PKG_LICENSE_FILES:=LICENSE
 
-PKG_BUILD_DEPENDS:=\
-	node/host \
-	reforis-distutils/host \
+HOST_PYTHON3_PACKAGE_BUILD_DEPENDS:=reforis-distutils
+PKG_BUILD_DEPENDS:=node/host
 
 include $(INCLUDE_DIR)/package.mk
 include $(TOPDIR)/feeds/packages/lang/python/python3-package.mk
diff --git a/web/reforis/reforis-storage-plugin/Makefile b/web/reforis/reforis-storage-plugin/Makefile
index 74f5c62d1..c50990154 100644
--- a/web/reforis/reforis-storage-plugin/Makefile
+++ b/web/reforis/reforis-storage-plugin/Makefile
@@ -20,9 +20,8 @@ PKG_MAINTAINER:=CZ.NIC <packaging@turris.cz>
 PKG_LICENSE:=GPL-3.0-or-later
 PKG_LICENSE_FILES:=LICENSE
 
-PKG_BUILD_DEPENDS:=\
-	node/host \
-	reforis-distutils/host \
+HOST_PYTHON3_PACKAGE_BUILD_DEPENDS:=reforis-distutils
+PKG_BUILD_DEPENDS:=node/host
 
 include $(INCLUDE_DIR)/package.mk
 include $(TOPDIR)/feeds/packages/lang/python/python3-package.mk
-- 
2.25.1

